hatchways/team-willow
# BE: Payments Controller

### What this PR does (required):
- Create and track payment flow with Stripe and db.

### Any information needed to test this feature (required):
- Ensure TOKEN environment variable is a valid encoded JWT for all tests.
- Run the following scripts in order:
1.  `npm run testPayment` <- Creates a payment document and paymentIntent.
      - Insert a valid id for `userId` and `sitterId` (register two users).
2. `npm run testConfirmPayment` <- Sets paid status.
     - Insert the `paymentIntentId` on line 10. Obtain from the payments collection or Stripe dashboard. 
3. `npm run testCancelIntent` <- Sets canceled status
     - Insert the paymentIntentId on line 10.

Note: Test 2 and 3 cannot be run sequentially.

### Any issues with the current functionality (optional):
- Doesn't discriminate between account types. Owners can cancel payments.
- Only tested with payments from cards.
